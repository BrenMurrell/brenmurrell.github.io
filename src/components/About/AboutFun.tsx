
import React from 'react';
import Card from "../Card/Card";

interface AboutFunProps {
    title?: string,
}

const AboutFun: React.FC<AboutFunProps> = ({
    title = 'What ado I do for fun?'
}) => {
    return (
        <>
            <h2>{title}</h2>
            <Card title="Front-end development" children={[<p>Front end details</p>]} />
            <Card title="Testing" children={[<p>Testing details</p>]} />
            <Card title="Accessibility" children={[<p>Accessibility details</p>]} />
            <Card title="Mentoring &amp; Leading" children={[<p>Mentoring</p>]} />
        </>
    );
};

export default AboutFun;