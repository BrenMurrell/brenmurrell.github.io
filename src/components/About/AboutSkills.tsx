
import React from 'react';
import Card from "../Card/Card";

interface AboutSkillsProps {
    title?: string,
}

const AboutSkills: React.FC<AboutSkillsProps> = ({
    title = 'What are my core skills?'
}) => {
    // Add your component logic here

    return (
        <>
            <h2>{title}</h2>
            <p>
                First and foremost, I am a skilled front-end practioner with over 15 years experience
                working in web development, and over 25 years of experience working in IT. Among my skills I
                pride myself on:
            </p>
            <ul>
                <li>Front-end development</li>
                <li>Testing</li>
                <li>Accessibility</li>
                <li>Mentoring and leading teams of developers</li>
            </ul>
            <Card title="Front-end development" children={[<p>Front end details</p>]} />
            <Card title="Testing" children={[<p>Testing details</p>]} />
            <Card title="Accessibility" children={[<p>Accessibility details</p>]} />
            <Card title="Mentoring &amp; Leading" children={[<p>Mentoring</p>]} />
        </>
    );
};

export default AboutSkills;