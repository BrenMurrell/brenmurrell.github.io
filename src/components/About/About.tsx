import Card from "../Card/Card";
import AboutSkills from "./AboutSkills";

const About = () => {
    // Your code here

    return (
        <>
            <h1>About me</h1>
            <p>
                Hi! I'm Bren, and I'm a Senior Front End Developer who is passionate about
                building robust, accessible software while empowering my teams to become
                strongly capable, compassionate developers.
            </p>
            <h2>What do I care about?</h2>
            <Card 
                title='People.'
                children={[<p>
                    As a developer, people are the heart of everything I do. I don't build software
                    for the sake of building software. I build software to help people. <em>All</em> people.
                    It's my duty as a developer to make sure that my deliverables are accessible to everyone through
                    empathy and a deep understanding of WCAG standards. It's also my duty to 
                    build and enable capable, diverse and inclusive teams that can deliver robust, thoughtfully designed
                    and rigoursouly tested projects.
                </p>]}
            />
            
            <AboutSkills />
        </>
    );
};

export default About;