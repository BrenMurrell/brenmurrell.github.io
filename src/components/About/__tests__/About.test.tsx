import { render } from "@testing-library/react";
import About from "../About";

describe('About', () => {
    it('renders' ,() => {
        const container = render(<About />);
        expect(container).toMatchSnapshot();
    })
});