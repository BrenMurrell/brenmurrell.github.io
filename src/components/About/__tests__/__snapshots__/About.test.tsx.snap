// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        About me
      </h1>
      <p>
        Hi! I'm Bren, and I'm a Senior Front End Developer who is passionate about building robust, accessible software while empowering my teams to become strongly capable, compassionate developers.
      </p>
      <h2>
        What do I care about?
      </h2>
      <h3>
        People.
      </h3>
      <p>
        As a developer, people are the heart of everything I do. I don't build software for the sake of building software. I build software to help people. 
        <em>
          All
        </em>
         people. It's my duty as a developer to make sure that my deliverables are accessible to everyone through empathy and a deep understanding of WCAG standards. It's also my duty to build and enable capable, diverse and inclusive teams that can deliver robust, thoughtfully designed and rigoursouly tested projects.
      </p>
      <h2>
        What are my core skills?
      </h2>
      <p>
        First and foremost, I am a skilled front-end practioner with over 15 years experience working in web development, and over 25 years of experience working in IT. Among my skills I pride myself on:
      </p>
      <ul>
        <li>
          Front-end development
          <ul>
            <li>
              Accessible HTML & CSS
            </li>
            <li>
              JavaScript - particularly React and Typescript
            </li>
          </ul>
        </li>
        <li>
          Mentoring and leading teams of developers
        </li>
        <li />
      </ul>
    </div>
  </body>,
  "container": <div>
    <h1>
      About me
    </h1>
    <p>
      Hi! I'm Bren, and I'm a Senior Front End Developer who is passionate about building robust, accessible software while empowering my teams to become strongly capable, compassionate developers.
    </p>
    <h2>
      What do I care about?
    </h2>
    <h3>
      People.
    </h3>
    <p>
      As a developer, people are the heart of everything I do. I don't build software for the sake of building software. I build software to help people. 
      <em>
        All
      </em>
       people. It's my duty as a developer to make sure that my deliverables are accessible to everyone through empathy and a deep understanding of WCAG standards. It's also my duty to build and enable capable, diverse and inclusive teams that can deliver robust, thoughtfully designed and rigoursouly tested projects.
    </p>
    <h2>
      What are my core skills?
    </h2>
    <p>
      First and foremost, I am a skilled front-end practioner with over 15 years experience working in web development, and over 25 years of experience working in IT. Among my skills I pride myself on:
    </p>
    <ul>
      <li>
        Front-end development
        <ul>
          <li>
            Accessible HTML & CSS
          </li>
          <li>
            JavaScript - particularly React and Typescript
          </li>
        </ul>
      </li>
      <li>
        Mentoring and leading teams of developers
      </li>
      <li />
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
